starlette/datastructures.py

```diff
@@ -274,33 +284,45 @@ def __init__(
         self._dict = {k: v for k, v in _items}
         self._list = _items
 
-    def getlist(self, key: typing.Any) -> typing.List[typing.Any]:
+    def getlist(self, key: typing.Any) -> typing.List[_ValueType]:
         return [item_value for item_key, item_value in self._list if item_key == key]
 
-    def keys(self) -> typing.KeysView:
+    def keys(self) -> typing.KeysView[_KeyType]:
         return self._dict.keys()
 
-    def values(self) -> typing.ValuesView:
+    def values(self) -> typing.ValuesView[_ValueType]:
         return self._dict.values()
 
-    def items(self) -> typing.ItemsView:
+    def items(self) -> typing.ItemsView[_KeyType, _ValueType]:
         return self._dict.items()
 
-    def multi_items(self) -> typing.List[typing.Tuple[str, str]]:
+    def multi_items(self) -> typing.List[typing.Tuple[_KeyType, _ValueType]]:
         return list(self._list)
 
-    def get(self, key: typing.Any, default: typing.Any = None) -> typing.Any:
+    @typing.overload
+    def get(self, key: _KeyType) -> _ValueType:
+        ...  # pragma: no cover
+
+    @typing.overload
+    def get(
+        self,
+        key: _KeyType,
+        default: typing.Optional[typing.Union[_ValueType, _DefaultType]] = ...,
+    ) -> typing.Union[_ValueType, _DefaultType]:
+        ...  # pragma: no cover
+
+    def get(self, key: _KeyType, default: typing.Any = None) -> typing.Any:
```

Comment:
https://docs.python.org/3/library/collections.abc.html#collections-abstract-base-classes

I'm not quite sure if there are other benefits to implementing mixin methods like `.get`. `MutableMapping` already implements them. And they are already of the correct type.

![image](https://user-images.githubusercontent.com/25364828/169959369-9f049329-6219-4ad1-a1c0-bebd33c41fa8.png)

Reply:
So we can just delete `get` here! Great point. This makes this a PR that (maybe) changes behavior instead of just typing, but I think that's worth it 😄

Reply:
Thank again for this, great catch!